src/main/java/database/service/service/JsonDatabaseService.java

```diff
@@ -0,0 +1,272 @@
+            throw new DeserializeDatabaseException("Unable to deserialize data.");
+        }
+
+        boolean isEntityRemoved = entities.removeIf(e -> id.equals(e.getId()));
+        if (isEntityRemoved) {
+            try {
+                objectMapper.writeValue(databasePath.toFile(), entities);
+            } catch (IOException e) {
+                throw new SerializeDatabaseException("Unable to serialize data.");
+            }
+        }
+
+        return isEntityRemoved;
+    }
+
+    @Override
+    public void removeAllRecordsFromTable(Class<? extends BaseEntity> entityClass) {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+        try {
+            Files.writeString(databasePath, EMPTY_BRACKETS_TO_JSON);
+        } catch (IOException e) {
+            throw new SerializeDatabaseException("Unable to serialize data.");
+        }
+    }
+
+    @Override
+    public <T extends BaseEntity, I> T getById(Class<? extends BaseEntity> entityClass, I id) {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+
+        List<T> entities;
+        try {
+            entities = deserializeEntities(entityClass, Files.readString(databasePath));
+        } catch (IOException e) {
+            throw new DeserializeDatabaseException("Unable to deserialize data.");
+        }
+
+        return entities.stream()
+                .filter(e -> id.equals(e.getId()))
+                .findFirst()
+                .orElseThrow(() -> new IdDoesNotExistException("Entity with provided Id does not exist."));
+    }
+
+    @Override
+    public <T extends BaseEntity> Iterable<T> getAllRecordsFromTable(Class<? extends BaseEntity> entityClass) {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+
+        try {
+            return deserializeEntities(entityClass, Files.readString(databasePath));
+        } catch (IOException e) {
+            throw new DeserializeDatabaseException("Unable to deserialize data.");
+        }
+    }
+
+    @Override
+    public <T extends BaseEntity, V> Iterable<T> getByFilters(Class<? extends BaseEntity> entityClass,
+                                                              Set<Pair<String, V>> filters) {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+        Field[] fields = entityClass.getDeclaredFields();
+        Validator validator = new Validator();
+
+        List<T> entities;
+        try {
+            entities = deserializeEntities(entityClass, Files.readString(databasePath));
+        } catch (IOException e) {
+            throw new DeserializeDatabaseException("Unable to deserialize data.");
+        }
+
+        return entities.stream()
+                .filter(entity -> {
+                    for (Pair<String, V> filter : filters) {
+                        String propertyName = filter.getKey();
+                        V expectedValue = filter.getValue();
+                        validator.validateDatabaseFilters(fields, filters);
```

Comment:
We don't need to make any actions if filters are incorrect. Please move validation to the top.